lmfit/minimizer.py

```diff
@@ -1497,34 +1493,38 @@ def emcee(self, params=None, steps=1000, nwalkers=100, burn=0, thin=1,
                                       handle_inf=False)
 
         # If uncertainty was automatically estimated, weight the residual properly
-        if (not is_weighted) and (result.residual.size > 1):
-            if '__lnsigma' in params:
-                result.residual = result.residual/np.exp(params['__lnsigma'].value)
+        if (
+            (not is_weighted)
+            and (result.residual.size > 1)
+            and '__lnsigma' in params
+        ):
+            result.residual = result.residual/np.exp(params['__lnsigma'].value)
 
         # Calculate statistics for the two standard cases:
         if isinstance(result.residual, np.ndarray) or (float_behavior == 'chi2'):
             result._calculate_statistics()
 
-        # Handle special case unique to emcee:
-        # This should eventually be moved into result._calculate_statistics.
         elif float_behavior == 'posterior':
-            result.ndata = 1
-            result.nfree = 1
+            self._extracted_from_emcee_201(result)
+        if auto_pool is not None:
+            auto_pool.terminate()
 
-            # assuming prior prob = 1, this is true
-            _neg2_log_likel = -2*result.residual
+        return result
 
-            # assumes that residual is properly weighted, avoid overflowing np.exp()
-            result.chisqr = np.exp(min(650, _neg2_log_likel))
+    # TODO Rename this here and in `emcee`
+    def _extracted_from_emcee_201(self, result):
+        result.ndata = 1
+        result.nfree = 1
 
-            result.redchi = result.chisqr / result.nfree
-            result.aic = _neg2_log_likel + 2 * result.nvarys
-            result.bic = _neg2_log_likel + np.log(result.ndata) * result.nvarys
+        # assuming prior prob = 1, this is true
+        _neg2_log_likel = -2*result.residual
 
-        if auto_pool is not None:
-            auto_pool.terminate()
+        # assumes that residual is properly weighted, avoid overflowing np.exp()
+        result.chisqr = np.exp(min(650, _neg2_log_likel))
 
-        return result
+        result.redchi = result.chisqr / result.nfree
+        result.aic = _neg2_log_likel + 2 * result.nvarys
+        result.bic = _neg2_log_likel + np.log(result.ndata) * result.nvarys
```

Comment:
I don't follow this as the diff is hard to read; what is the actual change here? Why do we need a new function `_extracted_from_emcee_201` - unless there is a convincing argument to do this, I am not really inclined to accept this.